lib/puppet/provider/foreman_smartproxy/rest_v3.rb

```diff
@@ -3,8 +3,19 @@
 
   def proxy
     @proxy ||= begin
-      r = request(:get, 'api/v2/smart_proxies', :search => %{name="#{resource[:name]}"})
-      raise Puppet::Error.new("Proxy #{resource[:name]} cannot be retrieved: #{error_message(r)}") unless success?(r)
+      path = 'api/v2/smart_proxies'
+      r = request(:get, path, :search => %{name="#{resource[:name]}"})
+
+      unless success?(r)
+        error_string = <<~ERROR.chomp
+
+          SmartProxy #{resource[:name]} cannot be retrieved from Foreman:
+          Error making GET request to #{request_uri(path)}
```

Comment:
```suggestion
          Error making GET request to #{r.uri)}
```

Reply:
I already tried this previously and it did not work. `#{r.uri}` would evaluate to an empty string in my tests. This is the reason I introduced the `request_uri` method.